executor/api/rest/client.go

```diff
@@ -252,6 +252,24 @@ func (smc *JSONRestClient) Metadata(ctx context.Context, modelName string, host
 	return smc.call(ctx, modelName, smc.modifyMethod(client.SeldonMetadataPath, modelName), host, port, msg, meta)
 }
 
+func (smc *JSONRestClient) ModelMetadata(ctx context.Context, modelName string, host string, port int32, msg payload.SeldonPayload, meta map[string][]string) (payload.ModelMetadata, error) {
```

Comment:
Conscious this is for consistency across all functions above, btu why are we retunring `payload.ModelMetadata` by value instead of reference/pointer? It seems like it would be quite inefficient to copy all the values that are returned by the client

Reply:
Good point, will adjust it.

Reply:
Actually, yeah, I did mostly for consistency with other methods. Not sure how I can make it better?

Reply:
Following on this in https://github.com/SeldonIO/seldon-core/issues/2014

Reply:
`payload.ModelMetadata` is actually an interface which can hold either a struct or a pointer to a struct. Here a pointer is being passed so no copying is taking place. Will update the issue #2014 with this information too.